Tests/tests/test_RDKShell.cpp

```diff
@@ -0,0 +1,649 @@
+   ON_CALL(rdkshellmock, removeKeyMetadataListener(::testing::_))
+                .WillByDefault(::testing::Return(false));
+
+   EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("removeKeyMetadataListener"), _T("{\"client\": \"org.rdk.Netflix\"}"), response));
+
+}
+
+TEST_F(RDKShellTest, injectKey)
+{
+   EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("injectKey"), _T("{}"), response));
+   
+   ON_CALL(rdkshellmock, injectKey(::testing::_, ::testing::_))
+                .WillByDefault(::testing::Return(false));
+   EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("injectKey"), _T("{\"keyCode\": 10,"
+                                                                                           "\"modifiers\": ["
+                                                                                                "\"shift\""
+                                                                                           "],"
+                                                                                    "}"), response));
+}
+
+TEST_F(RDKShellTest, getZOrder)
+{
+	EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getZOrder"), _T("{}"), response));
+}
+
+TEST_F(RDKShellTest, HolePunch)
+{
+  ON_CALL(rdkshellmock, setHolePunch(::testing::_, ::testing::_))
+                .WillByDefault(::testing::Return(true));
+  EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("setHolePunch"), _T("{\"client\": \"org.rdk.Netflix\","
+                                                                                             "\"callsign\": \"org.rdk.Netflix\","
+                                                                                             "\"holePunch\": true}"), response));
+  ON_CALL(rdkshellmock, getHolePunch(::testing::_, ::testing::_))
+            .WillByDefault(::testing::Invoke(
+                [](const std::string& client, bool& holePunch){
+                      holePunch = true;
+                      return true;
+                }));
+
+  EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getHolePunch"), _T("{\"client\": \"org.rdk.Netflix\","
+                                                                                             "\"callsign\": \"org.rdk.Netflix\"}"), response));
+
+
+}
+
+TEST_F(RDKShellTest, Scale)
+{
+  ON_CALL(rdkshellmock, getScale(::testing::_, ::testing::_, ::testing::_))
+            .WillByDefault(::testing::Invoke(
+                [](const std::string& client, double &scaleX, double &scaleY){
+                      scaleX = 0.5;
+		      scaleY = 0.5;
+                      return true;
+                }));
+
+  EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getScale"), _T("{\"client\": \"org.rdk.Netflix\","
+                                                                                             "\"callsign\": \"org.rdk.Netflix\"}"), response));
+
+
+}
+
```

Comment:
shouldn't you check JSON-RPC response in JSON-RPC tests?

Reply:
i will add that